pandas/io/pytables.py

```diff
@@ -1459,6 +1459,11 @@ def __init__(self, store, s, func, where, nrows, start=None, stop=None,
                 stop = nrows
             stop = min(nrows, stop)
 
+            # Piggy-back normalized `nrows` (considering start and stop) onto
+            # PyTables tables object so that the GenericIndexCol constructor
+            # knows how large the index should be.
+            self.s.table._nrows_to_read = stop - start
```

Comment:
Why not make these class attributes of `self` ?

Reply:
Huhu!

> start and stop - why not make these instance attributes of self ?

Hm, what do you have in mind about doing with them after having set them as instance attributes of `self`? In the three lines below we're doing exactly that (we set `self.start` and self.stop`; did you mean something else?)

Reply:
> Or even just compute nrows as an attribute?

Interesting. I thought about this a bit but I am not sure what you mean. Where; on the `PyTables` table object?

That would be quite invasive. A `PyTables` table object might use its `nrows` property internally. 

Adding a new private attribute to an object (guaranteed to not mess with the object's intrinsics) is not beautiful, but a little less invasive than overwriting an object property which might mess with that object's internal consistency.

Reply:
@jgehrcke : I'm talking about attaching a new attribute to the `TableIterator` instance itself.

Attaching attributes to the `TableIterator`'s `s.table` is riskier.

Reply:
> @jgehrcke : I'm talking about attaching a new attribute to the TableIterator instance itself.

Gotcha! I'll look into that. Thanks for the feedback, much appreciated. 

> Attaching attributes to the TableIterator's s.table is riskier.

Ack!

Reply:
There is however no obvious way for me to access the `TableIterator` instance from the `GenericIndexCol` instance. The latter inherits from `IndexCol` which gets its `.table` attribute set through a `set_table()` method, but as far as I can see there is no existing obvious relationship to reach back to the `TableIterator`... I think I had the problem before which is why I chose the table object to piggy-back the piece of information on.

Reply:
@gfyoung so this is the context where we call into the `convert()` method of the `GenericIndexCol`:

https://github.com/pandas-dev/pandas/blob/a60888ce4ce9e106537fb410688b66baa109edc3/pandas/io/pytables.py#L3437

There are certainly other ways to pass along the `_nrows_to_read` concept but I didn't see a cleaner way yet, specifically I didn't see a way that does not add additional scaffold, passing arguments around etc.

Did you have something very specific in mind that I simply didn't see?

Reply:
Interesting...we might have to then propagate an `nrows` parameter throughout the `IndexCol` instantiations.

@jreback : Thoughts?

Reply:
> Attaching a parameter to an object we have no information about is definitely riskier and hackier (and is therefore a practice I would discourage) compared to passing arguments around.

I do agree.